pkg/clioptions/clusterdiscovery/cluster.go

```diff
@@ -242,6 +244,16 @@ func LoadConfig(state *ClusterState) (*ClusterConfiguration, error) {
 			config.HasIPv4 = true
 		}
 	}
+	// IP order matters for DualStack.
+	for _, cidr := range state.NetworkSpec.ServiceNetwork {
```

Comment:
Can we have `if config.HasIPv6 && config.HasIPv4` before we run this loop? If this is something for dual-stack only, then let's run it only for dual-stack

Reply:
Will shorten comment.  This is for single and dual.

> Plus, as we are detecting primary stack based on the 1st network, would utilnet.IsIPv6CIDRString(state.NetworkSpec.ServiceNetwork[0]) suffice?

I was trying to think of a way to combine the loops.

Reply:
>This is for single and dual.

I miss something to understand how this added code is relevant for single-stack... For single-stack the previous loop (just above the code you add) should already correctly detect the IP stack, or no?

Reply:
The new loop replaces the old logic

```
// these constants are taken from kube e2e and used by tests
context.IPFamily = "ipv4"
if config.HasIPv6 && !config.HasIPv4 {
        context.IPFamily = "ipv6"
}
```

Reply:
That's fair, thanks for explaining!

Reply:
avoided loop by assuming IPv4 default.